cluster/gce/master.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A library with any master related operations during kube-up, kube-down etc.
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${KUBE_ROOT}/cluster/gce/common.sh"
+
+# Provision master machine and configures network to give access to it over HTTPS.
+function master::provision {
+  master::configure-network
+  master::create
+
+  # Wait for last batch of jobs
+  wait-for-jobs
+}
+
+# Configures network to give access to master machine over HTTPS.
+function master::configure-network {
+  detect-project
+
+  echo "Creating firewall rule for HTTPS traffic to ${MASTER_NAME}"
+  gcloud compute firewall-rules create "${MASTER_NAME}-https" \
+    --project "${PROJECT}" \
+    --network "${NETWORK}" \
+    --target-tags "${MASTER_TAG}" \
+    --allow tcp:443 &
+}
+
+# Creates master machine with a static IP and persistent disk.
+function master::create {
+  KUBE_MASTER=${MASTER_NAME}
+  master::create-disk
+  master::create-ip
+  master::create-instance "${KUBE_MASTER_IP}" &
+}
+
+# Creates persistent disk for master machine.
+function master::create-disk {
+  detect-project
```

Comment:
Can we do a `detect-project` call in `master::provision` instead of calling it multiple times in the subroutines?